app/javascript/utils/lazy-highlight-all.ts

```diff
@@ -0,0 +1,46 @@
+export function lazyHighlightAll(): void {
+  let highlighted = false
+
+  function loadAndHighlight() {
+    import('@/utils/highlight')
+      .then((m) => {
+        m.highlightAll()
+        observer.disconnect()
+      })
+      .catch((e) => {
+        observer.disconnect()
+        throw new Error(e)
+      })
+  }
+
+  function checkAndHighlight(): void {
+    // only load highlightjs when a code block exists in the DOM
+    if (highlighted || !hasHighlightableCodeBlock(document)) return
+
+    // to avoid race conditions, set this optimistically to true before the async loading
+    highlighted = true
+    loadAndHighlight()
+  }
+
+  function mutationCallback(mutationsList: MutationRecord[]): void {
+    for (const mutation of mutationsList) {
+      if (mutation.type !== 'childList') continue
+      for (const node of mutation.addedNodes) {
```

Comment:
```suggestion
      if (mutation.type !== 'childList') continue
      
      for (const node of mutation.addedNodes) {
```

Reply:
Always good to put a linebreak after guards.

Reply:
Adding those!